incubator/datadog-apm/templates/cluster-agent-deployment.yaml

```diff
@@ -0,0 +1,279 @@
+{{ toYaml .Values.clusterAgent.strategy | indent 4 }}
+{{- else }}
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 0
+    type: RollingUpdate
+{{- end }}
+  selector:
+    matchLabels:
+      app: {{ template "datadog-apm.fullname" . }}-cluster-agent
+        {{- if .Values.clusterAgent.podLabels }}
+{{ toYaml .Values.clusterAgent.podLabels | indent 6 }}
+        {{- end }}
+  template:
+    metadata:
+      labels:
+        app: {{ template "datadog-apm.fullname" . }}-cluster-agent
+        {{- if .Values.clusterAgent.podLabels }}
+{{ toYaml .Values.clusterAgent.podLabels | indent 8 }}
+        {{- end }}
+      name: {{ template "datadog-apm.fullname" . }}-cluster-agent
+      annotations:
+        checksum/install_info: {{ printf "%s-%s" .Chart.Name .Chart.Version | sha256sum }}
+        {{- if .Values.clusterAgent.datadog_cluster_yaml }}
+        checksum/clusteragent-config: {{ tpl (toYaml .Values.clusterAgent.datadog_cluster_yaml) . | sha256sum }}
+        {{- end }}
+        {{- if .Values.clusterAgent.confd }}
+        checksum/confd-config: {{ tpl (toYaml .Values.clusterAgent.confd) . | sha256sum }}
+        {{- end }}
+        ad.datadoghq.com/cluster-agent.check_names: '["prometheus"]'
+        ad.datadoghq.com/cluster-agent.init_configs: '[{}]'
+        ad.datadoghq.com/cluster-agent.instances: |
+          [{
+            "prometheus_url": "http://%%host%%:5000/metrics",
+            "namespace": "datadog.cluster_agent",
+            "metrics": [
+              "go_goroutines", "go_memstats_*", "process_*",
+              "api_requests",
+              "datadog_requests", "external_metrics", "rate_limit_queries_*",
+              "cluster_checks_*"
+            ]
+          }]
+      {{- if .Values.clusterAgent.podAnnotations }}
+{{ toYaml .Values.clusterAgent.podAnnotations | indent 8 }}
+      {{- end }}
+
+    spec:
+      {{- if .Values.clusterAgent.priorityClassName }}
+      priorityClassName: "{{ .Values.clusterAgent.priorityClassName }}"
+      {{- end }}
+      {{- if .Values.clusterAgent.image.pullSecrets }}
+      imagePullSecrets:
+{{ toYaml .Values.clusterAgent.image.pullSecrets | indent 8 }}
+      {{- end }}
+      serviceAccountName: {{ if .Values.clusterAgent.rbac.create }}"{{ template "datadog-apm.fullname" . }}"{{ else }}"{{ .Values.clusterAgent.rbac.serviceAccountName }}"{{ end }}
+      {{- if .Values.clusterAgent.useHostNetwork }}
+      hostNetwork: {{ .Values.clusterAgent.useHostNetwork }}
+      dnsPolicy: ClusterFirstWithHostNet
+      {{- end }}
+      {{- if .Values.clusterAgent.dnsConfig }}
+      dnsConfig:
+{{ toYaml .Values.clusterAgent.dnsConfig | indent 8 }}
+      {{- end }}
+      containers:
+      - name: cluster-agent
+        image: "{{ .Values.clusterAgent.image.repository }}:{{ .Values.clusterAgent.image.tag }}"
+        {{- with .Values.clusterAgent.command }}
+        command: {{ range . }}
+          - {{ . | quote }}
+        {{- end }}
```

Comment:
I think we can hard code the commands we know should run in this chart ('["trace-agent", "-config=/etc/datadog-agent/datadog.yaml"]`) and use the values provided in the `values.yml` file to add extra args if necessary.